### PR TITLE
Combine Start Work & Project Creation Forms

### PR DESCRIPTION
**Pull request by zoek1**
##### Description
- [X] Build the form to capture the two scenarios outlined in the mockups (hacking solo, hacking as a team).
- [X]  Allow users to designate whether they are looking to accept team members
 On looking for a skill, we should pre populate the tags that we use on the funder form (front end, backend, design, etc)
- [X]  Project name, logo, summary, plan, repo should be editable
until you submit work
- [X]  Hackers should be able to edit their project details via that same modal

DEMO:
https://www.loom.com/share/430a1a6963ee487ab4f3c580d5b6c64d

##### Refers/Fixes
https://github.com/gitcoinco/web/issues/6444

##### Testing
image

image

ETC bounties
https://www.loom.com/share/d68910dfeadd4c7b9858aff2419358ee

### CODE REVIEW

**Comment by PixelantDesign** [comment]
Looking good!

**Comment by thelostone-mc** [comment]
@zoek1 could you throw in video to help @PixelantDesign validate the flow ?

**Comment by zoek1** [comment]
I just posted the video demo in the descriptions @PixelantDesign @thelostone-mc :)

**Comment by PixelantDesign** [comment]
looks great!

**Comment by PixelantDesign** [comment]
@alexvotofuture @connoroday what do you think?

**Comment by alexvotofuture** [comment]
Awesome, @zoek1. Looks like it functions exactly as we expected. Thank you for the work on this!

File: app/assets/v2/js/pages/bounty_details.js [hunk 1]
```diff
@@ -699,20 +699,64 @@ var attach_override_status = function() {
...
+      if (document.result.event) {
+        $(document).on('change', '#project_logo', function() {
+          previewFile($(this));
+        });
+        projectForm.on('submit', function(e) {
+          e.preventDefault();
+          let logo = $(this)[0]['logo'].files[0];
+          let data = $(this).serializeArray();
+
+          submitInterest(document.result['pk'], '', self, () => {
```
Comment (octavioamu) [inline app/assets/v2/js/pages/bounty_details.js:751]: I believe this will be an empty interest on the activity.

**Comment by thelostone-mc** [comment]
@zoek once you address @octavioamu comments ~ Could you throw in a demo for hackathons bounty creation flow for ETC 
Add ETC as a mainnet token via http://localhost:8000/_administrationeconomy/token/ -> create an ETC bounty and make sure the start + project creation works

File: app/assets/v2/js/pages/bounty_details2.js [hunk 2]
```diff
@@ -601,7 +601,12 @@ var show_interest_modal = function() {
     modalBody.load(modalUrl, ()=> {
       let actionPlanForm = $('#action_plan');
       let issueMessage = $('#issue_message');
+      let data = $('.team-users').data('initial') ? $('.team-users').data('initial').split(', ') : [];
```
[the remark this answers is not in the document]
Comment (zoek1) [reply app/assets/v2/js/pages/bounty_details2.js:643]: https://www.loom.com/share/d68910dfeadd4c7b9858aff2419358ee

File: app/assets/v2/js/pages/bounty_details.js (hunk 1 above)
Comment (zoek1) [reply app/assets/v2/js/pages/bounty_details.js]: Replaced the empty string with the project description

**Comment by PixelantDesign** [comment]
Thanks for the video @zoek1! I see the ETC bounty there!

File: app/dashboard/models.py [hunk 3]
```diff
@@ -4716,6 +4716,7 @@ class HackathonProject(SuperModel):
     )
     looking_members = models.BooleanField(default=False)
     chat_channel_id = models.CharField(max_length=255, blank=True, null=True)
+    extra = JSONField(default=dict, blank=True, null=True)
```
Comment (thelostone-mc) [inline app/dashboard/models.py:4719]: could you add in helpt text to explain this better In a follow up PR

File: app/assets/v2/js/pages/bounty_details2.js [hunk 4]
```diff
@@ -619,8 +640,35 @@ var show_interest_modal = function() {
...
+        $(document).on('change', '#project_logo', function() {
+          previewFile($(this));
+        });
+        projectForm.on('submit', function(e) {
+          e.preventDefault();
+          let elements = $(this)[0];
+          let logo = elements['logo'].files[0];
+          let summary = elements['summary'].value;
+          let data = $(this).serializeArray();
+
```
Comment (thelostone-mc) [inline app/assets/v2/js/pages/bounty_details2.js:661]: ^ update to const

Comment (zoek1) [reply app/assets/v2/js/pages/bounty_details2.js:661]: done :)

File: app/dashboard/templates/addinterest.html [hunk 5]
```diff
@@ -99,6 +95,78 @@ <h5 class="text-center font-title">{% trans "Submit a Plan" %}</h5>
             </div>
             <button id="submit" class="row button button--primary btn-interested" type="submit">{% trans "Next" %}</button>
           </form>
+        {% elif is_registered and bounty.event %}
```
Comment (octavioamu) [inline app/dashboard/templates/addinterest.html:98]: so this message appear to people when they are not register yet to the hackathon but now is also showing the form behind. is that right? doesn't people need to go to onboard register and come back?

**Pull request merged**